torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -286,6 +286,9 @@ inline void errorIfCapturingNonCapturableNCCL(c10::cuda::CaptureStatus status) {
 const int64_t ProcessGroupNCCL::kWatchdogThreadSleepMillis = 1000;
 constexpr int64_t kSynchronizeBusyWaitMillis = 10;
 thread_local uint64_t ProcessGroupNCCL::ncclActiveGroupCounter_ = 0;
+std::mutex ProcessGroupNCCL::allProcessGroupsMutex_;
```

Comment:
I assume someone did a careful review of the brand new locking here. I haven't looked in details.